System.IO.FileSystem/nanoFramework/SDCard.cs

```diff
@@ -0,0 +1,241 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System.Runtime.CompilerServices;
+using Diagnostics = System.Diagnostics;
+
+namespace nanoFramework.System.IO.FileSystem
+{
+    /// <summary>
+    /// Class to allow a SD memory card to be configured and mounted on the system.
+    /// </summary>
+    public class SDCard
+    {
+
+#pragma warning disable 0649
+
+        // this field is set in native code
+        [Diagnostics.DebuggerBrowsable(Diagnostics.DebuggerBrowsableState.Never)]
+        private bool _mounted = false;
+
+#pragma warning restore 0649
+
+        private SDInterfaceType _sdCardType;
+        private bool _isCardDetectEnabled = false;
+        private int  _cardDetectPin;
+
+        #region Properties
+        /// <summary>
+        /// Type of interface used by SDCard.
+        /// </summary>
+        public SDInterfaceType CardType => _sdCardType;
+
+        /// <summary>
+        /// Indicates if the SD card has been mounted
+        /// </summary>
+        public bool IsMounted => _mounted;
+
+        /// <summary>
+        /// Indicates if SD card has been detected if optional cardDetectPin parameter is enabled with a valid GPIO pin.
+        /// If not enabled will always return false.
+        /// </summary>
+        /// <remarks>
+        /// Not all SD Card modules have a card detect pin or the pin connected to a GPIO pin. 
+        /// </remarks>
+        public bool IsCardDetected
+        {
+            get
+            {
+                if (_isCardDetectEnabled)
+                    return false;
+
+                return PollCardDetectNative(_cardDetectPin);
+            }
+        }
+        #endregion
+
+        #region Parameters
+        /// <summary>
+        /// Parameter used for creating a MMC card instance.
+        /// </summary>
+        public struct SDCardMmcParameters
+        {
+            /// <summary>
+            /// Data width to use on MMC SD protocol.
+            /// </summary>
+            public SDDataWidth dataWidth;
+
+            /// <summary>
+            /// Set true when an Card Detect Pin is used. 
+            /// The cardDetectPin parameter must have a valid GPIO pin.
+            /// </summary>
+            /// <remarks>
+            /// Not all SD Card modules have a card detect pin or the pin connected to a GPIO pin. 
+            /// </remarks>
+            public bool enableCardDetectPin;
+
+            /// <summary>
+            /// The optional card detect pin when pulled low indicates card is present in slot.
```

Comment:
1. Better "extra clarify" by saying that this expects a GPIO number.
2. Don't think there a need to state that the pin has to be pulled low or high. I mean that's entirely with the system designer and the native code can handle one or the other as needed.

Reply:
Done